Source: kde-packager@
Upstream: Yes, on 4.5, 4.6 and trunk branches.
Reason: Due to the way memcpy() is implemented in some newer glibc [1],
	users may experience crashes in the KDE imap4 ioslave.

From 99c7981115ed32d08ef89164bcea1afb0b9137dd Mon Sep 17 00:00:00 2001
From: winterz <winterz@283d02a7-25f6-0310-bc7c-ecb5cbfe19da>
Date: Fri, 21 Jan 2011 17:05:56 +0000
Subject: [PATCH] backport SVN commit r1216161 by winterz:
 Avoid overlapping memcpy in myRead()

git-svn-id: svn+ssh://svn.kde.org/home/kde/branches/KDE/4.6/kdepimlibs@1216163 283d02a7-25f6-0310-bc7c-ecb5cbfe19da
---
 kioslave/imap4/imap4.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/kioslave/imap4/imap4.cpp b/kioslave/imap4/imap4.cpp
index f020130..fb4892d 100644
--- a/kioslave/imap4/imap4.cpp
+++ b/kioslave/imap4/imap4.cpp
@@ -2576,7 +2576,7 @@ ssize_t IMAP4Protocol::myRead(void *data, ssize_t len)
     ssize_t copyLen = (len < readBufferLen) ? len : readBufferLen;
     memcpy(data, readBuffer, copyLen);
     readBufferLen -= copyLen;
-    if (readBufferLen) memcpy(readBuffer, &readBuffer[copyLen], readBufferLen);
+    if (readBufferLen) memmove(readBuffer, &readBuffer[copyLen], readBufferLen);
     return copyLen;
   }
   if (!isConnected()) return 0;
-- 
1.7.3.5

